### PR TITLE
function added for finding valid conditions for perturbed systems wit…

### PR DESCRIPTION
**Pull request by SabineAuer**
…h mass matrices

### CODE REVIEW

File: Project.toml [hunk 1]
```diff
@@ -5,6 +5,7 @@ version = "2.2.0"
 
 [deps]
 DiffEqBase = "2b5f629d-d688-5b77-993f-72d75c75574e"
+DifferentialEquations = "0c46a032-eb83-5123-abaf-570d42b7fbaa"
```
Comment (janlisse) [inline Project.toml]: This should not be necessary. We should use OrdinaryDiffEq only for now.

File: src/operationpoint/find_valid_initial_condition.jl [hunk 2]
```diff
@@ -0,0 +1,35 @@
...
+    nl_res = nlsolve(rr, ic_guess)
+    if converged(nl_res) == true
+        return nl_res.zero #State(pg, nl_res.zero)
+    else
+        println("Failed to find initial conditions on the constraint manifold!")
+        println("Try running nlsolve with other options.")
+    end
+end
+
+export RootRhs_ic
```
Comment (janlisse) [inline src/operationpoint/find_valid_initial_condition.jl]: Is it possible to write a unit test for this?

File: test/simulations/NodeShortCircuit.jl [hunk 3]
```diff
@@ -27,18 +26,16 @@ stat_lines = [
...
   StaticLine(;from=1, to=2, Y = 1/(0.1152 + im*0.0458))]
 
-nsc = NodeShortCircuit(R = 0.1, node = 1, sc_timespan=(1.,2.))
+nsc = NodeShortCircuit(node = 1, R = 0.1, sc_timespan=(1.,2.))
 
 pg_static = PowerGrid(busses, stat_lines)
 pg = PowerGrid(busses, lines)
 
-@test nsc(pg_static) == nothing
+#@test nsc(pg_static) == nothing
```
Comment (janlisse) [inline test/simulations/NodeShortCircuit.jl]: Can you delete this line instead of uncommenting?

File: Project.toml (hunk 1 above)
Comment (SabineAuer) [reply Project.toml]: @janlisse : I needed the package to access ODEFunction [here](https://github.com/JuliaEnergy/PowerDynamics.jl/pull/54/files#diff-d3c102d899631c26c9a6dae5a01633f4R16) and extract the mass matrix. Do you see an alternative here?

File: test/simulations/NodeShortCircuit.jl (hunk 3 above)
Comment (SabineAuer) [reply test/simulations/NodeShortCircuit.jl]: Yes, I can do that. No problem.

**Comment by SabineAuer** [comment]
@janlisse : do you know why code coverage is giving an error in the checks?

File: Project.toml (hunk 1 above)
Comment (janlisse) [reply Project.toml]: Yes, you can just use ```using OrdinaryDiffEq: ODEFunction```

File: test/simulations/NodeShortCircuit.jl (hunk 3 above)
Comment (janlisse) [reply test/simulations/NodeShortCircuit.jl]: Thanks :)

File: src/operationpoint/find_valid_initial_condition.jl (hunk 2 above)
Comment (janlisse) [reply src/operationpoint/find_valid_initial_condition.jl]: ok!

File: Project.toml (hunk 1 above)
Comment (SabineAuer) [reply Project.toml]: ok, done :)

**Pull request merged**